--- a/arch/arm/boot/dts/qcom/qcom-msm8974.dtsi	2026-09-10 12:06:52.550364214 +0400
+++ b/arch/arm/boot/dts/qcom/qcom-msm8974.dtsi	2026-09-10 12:06:52.560771562 +0400
@@ -813,8 +813,8 @@
 					wifi {
 						compatible = "qcom,wcnss-wlan";
 
-						interrupts = <GIC_SPI 145 IRQ_TYPE_EDGE_RISING>,
-							     <GIC_SPI 146 IRQ_TYPE_EDGE_RISING>;
+						interrupts = <GIC_SPI 145 IRQ_TYPE_LEVEL_HIGH>,
+							     <GIC_SPI 146 IRQ_TYPE_LEVEL_HIGH>;
 						interrupt-names = "tx", "rx";
 
 						qcom,smem-states = <&apps_smsm 10>, <&apps_smsm 9>;
